heron/stmgr/tests/cpp/server/stmgr_unittest.cpp

```diff
@@ -1521,7 +1521,8 @@ TEST(StMgr, test_tmaster_restart_on_same_address) {
   // Note: Here we sleep longer compared to the previous test as we need
   // to tmasterClient could take upto 1 second (specified in test_heron_internals.yaml)
   // to retry connecting to tmaster.
-  sleep(3);
+  while (regular_stmgr->GetPhysicalPlan()->stmgrs(1).data_port() == common.stmgr_baseport_ + 1)
```

Comment:
Can you give a reasonable timeout? As in the current code, 3-second was supposed to be reasonable, but it is not... So I don't know what is a reasonable one.